lib/binary-parser.js

```diff
@@ -0,0 +1,95 @@
+export class BinaryParser {
+  constructor() {
+    this.schema = [];
+  }
+
+  string(fieldName, options) {
+    this.schema.push({type: 'string', fieldName, ...options});
+    return this;
+  }
+
+  int(fieldName, options) {
+    this.schema.push({type: 'int', fieldName, ...options});
+    return this;
+  }
+
+  buffer(fieldName, options) {
+    this.schema.push({type: 'buffer', fieldName, ...options});
+    return this;
+  }
+
+  array(fieldName, options) {
+    this.schema.push({type: 'array', fieldName, ...options});
+    return this;
+  }
+
+  parse(buffer, parentContext) {
+    const view = new DataView(buffer.buffer);
```

Comment:
Be careful with this ... if the underlying buffer overruns the dimensions of `buffer`, this will read into garbage data. I think this needs to be:

```suggestion
    const view = new DataView(buffer.buffer, 0, buffer.byteLength);
```

Or maybe `buffer.length` if we know it's a `Uint8Array` ... or something like that.